mne/io/egi/egimff.py

```diff
@@ -0,0 +1,314 @@
+
+    time_n = dateutil.parser.parse(mff_hdr['date'])
+    info = dict(   # leer del info.xml desde read_mff_header
+        version=version,  # duda
+        year=int(time_n.strftime('%Y')),           # duda
+        month=int(time_n.strftime('%m')),          # duda
+        day=int(time_n.strftime('%d')),
+        hour=int(time_n.strftime('%H')),
+        minute=int(time_n.strftime('%M')),
+        second=int(time_n.strftime('%S')),
+        millisecond=int(time_n.strftime('%f')),
+        samp_rate=mff_hdr['Fs'],
+        n_channels=mff_hdr['nChans'],
+        gain=0,           # duda
+        bits=0,           # duda
+        value_range=0)    # duda
+    unsegmented = 1 if mff_hdr['nTrials'] == 1 else 0
+    precision = 4
+    # np.bitwise_and(version,6)
+    if precision == 0:
+        RuntimeError('Floating point precision is undefined.')
+    if unsegmented:
+        info.update(dict(n_categories=0,
+                         n_segments=1,
+                         n_samples=mff_hdr['nSamples'],
+                         n_events=0,
+                         event_codes=[],
+                         category_names=[],
+                         category_lengths=[],
+                         pre_baseline=0))
+    else:
+        raise NotImplementedError('Only continuos files are supported')
+    info['unsegmented'] = unsegmented
+    info['dtype'], info['orig_format'] = {2: ('>i2', 'short'),
+                                          4: ('>f4', 'float'),
+                                          6: ('>f8', 'double')}[precision]
+    info['dtype'] = np.dtype(info['dtype'])
+    return info, mff_hdr
+
+
+def _read_events(input_fname, hdr, info):
+    """Read events for the record.
+
+    in:
+        input_fname : str with the file path
+        hdr : dictionary with the headers get
+              by read_mff_header
+        info : header info array
+    """
+    mff_events, event_codes = read_mff_events(input_fname, hdr)
+    info['n_events'] = len(event_codes)
+    info['event_codes'] = np.asarray(event_codes).astype('<U4')
+    events = np.zeros([info['n_events'],
+                      info['n_segments'] * info['n_samples']])
+    for n, event in enumerate(event_codes):
+        for i in mff_events[event]:
+            events[n][i] = 1.4012984643248171e-45
+    return events, info
+
+
+def _combine_triggers_mff(data, remapping=None):
+    """Combine binary triggers."""
+    new_trigger = np.zeros(data.shape[1])
+    if data.astype(bool).sum(axis=0).max() > 1:  # ensure no overlaps
+        logger.info('    Found multiple events at the same time '
+                    'sample. Cannot create trigger channel.')
+        return
+    if remapping is None:
+        remapping = np.arange(data) + 1
+    for d, event_id in zip(data, remapping):
+        idx = d.nonzero()
+        if np.any(idx):
+            new_trigger[idx] += event_id
+    return new_trigger
+
+
+@verbose
+def read_raw_egi_mff(input_fname, montage=None, eog=None, misc=None,
```

Comment:
why not factorize this with the read_raw_egi to based on extension decide if you use mff or not?